pkg/preflight/run.go

```diff
@@ -255,14 +260,40 @@ func RunPreflights(interactive bool, output string, format string, args []string
 	stopProgressCollection()
 	progressCollection.Wait()
 
+	if len(analyzeResults) == 0 {
+		return types.NewExitCodeError(constants.EXIT_CODE_CATCH_ALL, errors.New("no data has been collected"))
+	}
+
+	exitCode := checkOutcomesToExitCode(analyzeResults)
+
 	if interactive {
-		if len(analyzeResults) == 0 {
-			return errors.New("no data has been collected")
+		err = showInteractiveResults(preflightSpecName, output, analyzeResults)
+	}
+
+	err = showTextResults(format, preflightSpecName, output, analyzeResults)
+
+	return types.NewExitCodeError(exitCode, err)
```

Comment:
You should not return an error if there is no error (exit code is 0)

Reply:
I think I've got a good enough way to do this now 👍